### ISSUE TITLE
After page refresh, symbols should be inserted at right z-index

### ISSUE THREAD

**Issue by mithunshanbhag** [cloudskew/cloudskew#94]
**Describe the bug**
This one is a little hard to explain on paper. 

Assume we have the following three symbols on the diagram canvas:
* symbol-1, with z-index = 0
* symbol-2, with z-index = 1
* symbol-3, with z-index = 2
 
Now let's delete symbol-2 (which has z-index = 1). So the canvas will now look like:
* symbol-1, with z-index = 0
* symbol-3, with z-index = 2

Now add a new symbol-4, it'll have z-index = 3. The canvas will now look like:
* symbol-1, with z-index = 0
* symbol-3, with z-index = 2
* symbol-4, with z-index = 3

Now refresh the page (F5) and add a new symbol-5. Weirdly, it'll have z-index = 1 (it'll occupy the 
'hole' left behind by deleted symbol-2 which had z-index = 1). The canvas will now look like:
* symbol-1, with z-index = 0
* symbol-5, with z-index = 1
* symbol-3, with z-index = 2
* symbol-4, with z-index = 3

Adding another new symbol-6 will give it z-index = 4. Canvas will look like:  
* symbol-1, with z-index = 0
* symbol-5, with z-index = 1
* symbol-3, with z-index = 2
* symbol-4, with z-index = 3
* symbol-6, with z-index = 4

In summary: After a page-refresh, the max recorded z-index is forgotten and new symbols will be inserted into "holes" (if any). If all "holes" are filled, then max recorded z-index is used.

**Solution**
We'll override the default behavior to always use max recorded z-index while inserting nodes, connectors or groups (irrespective of whether page has been refreshed or not).

**Comment by mithunshanbhag** [comment]
Fix is in pre-production (staging environment). Will be pushed to prod after testing (in a day or two).

CC: @arionmiles
